### PR TITLE
Move inline feedback form, update to pattern lib v1.6.2

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?

Pattern library v1.6.0 contains refactoring of the `guidance`, `definition` and `accordion` components. `guidance` has been removed, `definition` & `accordion` code base has changed to use a single `collapsible` module. The mark up changes required have been made in runner and tests updated.

The existing inline feedback form has moved to above the footer. The codebase for the form has been moved out of runner and into the pattern library so it can be shared by other services. An endpoint can now be specified as a 'data' attribute on the submit `button` - `data-endpoint="/foo"`. 

PL v1.6.0 also contains changes to the mutually exclusive input component to extend it's functionality. A small change was required to the existing mark up, changing `js-exclusive-checkbox-group` to `js-exclusive-group`.

### How to review 

Check the feedback form works as it did before. Screenshot below shows the new implementation ("Is there anything wrong with this page?"):
image

Review the existing question definitions and answer guidance components and check they work as they should.

Check the mutually exclusive checkboxes work.

### Checklist

* [ ] Inline feedback form works as expected.
* [ ] Answer guidance works as expected.
* [ ] Question definition works as expected.
* [ ] Mutually exclusive checkboxes works as expected.
* [ ] Tests pass.

### CODE REVIEW

**Comment by boxadesign** [comment]
The travis build is failing on a census functional test but all tests pass locally. Any ideas anyone?

**Comment by jonnyshaw89** [comment]
Is this gap suppose to be this big?
screenshot 2018-10-01 at 12 44 52

**Comment by MebinAbraham** [comment]
To me it looks like part of the survey, I know content makes it obvious but font and style is the same page. Just my view.

**Review by MebinAbraham — COMMENTED** [review]
I don't know if you can see it but the text is very slightly cut off on the bottom.

screen shot 2018-10-01 at 13 00 25

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
Is the `Show` button not longer meant to be displayed for definitions on devices smaller than 500px in width?

*Before:*
screen shot 2018-10-01 at 13 04 46

*After:* 
screen shot 2018-10-01 at 13 04 55

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
When you use Tab navigation, it goes from the `Show all` button straight to the feedback link. It should go to the next definition, not jump to the bottom?

File: app/templates/partials/feedback/expandable_inline.html [hunk 1]
```diff
@@ -1,6 +1,10 @@
-<div class="feedback__inline js-feedback-inline is-collapsed">
-    <div class="feedback__border">
-        {% include theme('partials/feedback_form.html') %}
-        <button class="btn btn--secondary feedback__cancel js-feedback-cancel" data-qa="a-feedback-close" {{helpers.track('click', 'Feedback', 'Cancel feedback link click')}}>{{ _("I don’t want to provide feedback") }}</button>
+<div id="feedback-thanks" class="js-form-confirmation u-hidden venus">Thanks for your feedback.</div>
```
Comment (MebinAbraham) [inline app/templates/partials/feedback/expandable_inline.html]: All new static content new to be marked for translation. `{{ _(My content) }}`

**Comment by MebinAbraham** [comment]
So many diff changes, definition, guidance, feedback etc. Not easy to thoroughly test all the html/js changes, maybe its just me. 

One of them things I guess, I understand its much easier to bring in the changes together. Maybe the feedback should be 1 PR and upgrading pattern lib and fixing breaking changes another PR. (In future anyway)

**Comment by boxadesign** [comment]
@jonnyshaw89 Yes the gap is meant to be that big. It prevents the feedback element getting confused with other in-context elements.

**Comment by boxadesign** [comment]
@MebinAbraham 

> To me it looks like part of the survey, I know content makes it obvious but font and style is the same page. Just my view.

- It's a fair enough view. We tried to minimise any potential confusion by creating a larger gap between the footer area and content so hopefully it'll be OK.

**Comment by boxadesign** [comment]
@MebinAbraham Can you show me this on your screen? I am unable to replicate this across browsers. It's possible it's to do with your monitor but it definitely shouldn't happen. I'll raise it in the PL as an issue anyway.

image

**Comment by boxadesign** [comment]
@MebinAbraham 

> So many diff changes, definition, guidance, feedback etc. Not easy to thoroughly test all the html/js changes, maybe its just me.
> 
> One of them things I guess, I understand its much easier to bring in the changes together. Maybe the feedback should be 1 PR and upgrading pattern lib and fixing breaking changes another PR. (In future anyway)

Yeah I agree this isn't ideal. The feedback change was what created this refactor, as all the similar components in the PL got change to use the single component.

File: app/templates/partials/feedback/expandable_inline.html (hunk 1 above)
Comment (boxadesign) [reply app/templates/partials/feedback/expandable_inline.html]: Yep missed that one

**Comment by boxadesign** [comment]
@MebinAbraham re: show button...

Yes it's meant to be hidden below `500px` - it was meant to do this already but the class to handle it wasn't working so it's now fixed.

**Comment by boxadesign** [comment]
@MebinAbraham RE: Tab Nav - I cannot replicate this issue.

File: app/templates/layouts/_questionnaire.html [hunk 2]
```diff
@@ -39,8 +37,10 @@
         <button class="btn btn--secondary js-btn-save" data-qa="btn-save-sign-out" type="submit" name="action[save_sign_out]" {{helpers.track('click', 'Navigation', 'Save and complete later click')}}>{% block save_sign_out_button_text %}{{ _("Save and complete later") }}{% endblock %}</button>
       </div>
     </form>
-
 {% endblock main %}
 {% block footer %}
+<div class="container u-mt-xl u-pt-l">
+{% include theme('partials/feedback/expandable_inline.html') %}
+</div>
```
Comment (jonnyshaw89) [inline app/templates/layouts/_questionnaire.html:44]: I think this has now been moved outside of the form? so now when we submit we get a HTTP 401 as its missing the csrf token

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/layouts/_questionnaire.html (hunk 2 above)
Comment (boxadesign) [reply app/templates/layouts/_questionnaire.html:44]: The form data wasn't getting sent so I've had to make a fix in the pattern library -> https://github.com/ONSdigital/sdc-global-design-patterns/pull/170

(Feel free to review the code)

**Comment by boxadesign** [comment]
@MebinAbraham @jonnyshaw89 PR has been updated with required changes and fixes

**Review by jonnyshaw89 — APPROVED** [review]

**Review by MebinAbraham — CHANGES_REQUESTED** [review]
With no JS, this happens on hover:
screen shot 2018-10-04 at 16 05 44

**Review by MebinAbraham — COMMENTED** [review]
So with no JS, you open a new page but not sure if we should say?:
`It is now safe to close this window.`
Because it opens in the same window and not in a new tab, hence they would close the survey?.

**Review by MebinAbraham — COMMENTED** [review]
The issue regarding feedback after submission can be looked at another time. #1809

**Comment by boxadesign** [comment]
@MebinAbraham Have fixed the non JS hover bug and also set the link to open a new tab.

File: app/templates/partials/summary/collapsible-summary.html [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
             {%- set outer_loop = loop -%}
-                <h3 class="accordion__title js-accordion-title u-cf icon--accordion" data-js-accordion-event-label="{{group.title}}">
-                    <span class="accordion__title-text">{{group.title}}</span>
-                    <button class="btn btn--secondary btn--small accordion__title-right accordion-unhide@m js-accordion-toggle u-wa--@xs" id="{{ group.id }}-button" data-close-label="{{ _("Hide") }}" data-open-label="{{ _("Show") }}" aria-hidden="true">{{ _("Show") }}</button>
+                <h3 class="collapsible__title js-collapsible-title u-cf icon--collapsible" data-ga="click" data-ga-category="Preview Survey" data-ga-action="Open panel" data-ga-label="{{group.title}}">
+                    <span class="collapsible__title-text">{{group.title}}</span>
+                    <button class="btn btn--secondary btn--small collapsible__title-right js-collapsible-toggle u-wa--@xs" id="{{ group.id }}-button" data-close-label="{{ _('Hide') }}" data-open-label="{{ _('Show') }}" aria-hidden="true" data-ga="click" data-ga-category="Preview Survey" data-ga-action="Open panel" data-ga-label="{{group.title}}">{{ _("Show") }}</button>
```
Comment (MebinAbraham) [inline app/templates/partials/summary/collapsible-summary.html:12]: My petty comment: we should be consistent. ie some are `{{ 'Show' }}` some are `{{ "Show" }}`. 

Also some files don't have new lines at the end now, nice to keep it all same.

**Review by MebinAbraham — APPROVED** [review]
All seems to be working as expected. 👍

**Comment by jonnyshaw89** [comment]
@boxadesign if you squash these commits we can get this in

**Pull request merged**